quipucords/scanner/network/connect.py

```diff
@@ -254,19 +266,25 @@ def _handle_ssh_passphrase(credential):
             pass
 
 
-def construct_connect_inventory(hosts, credential, connection_port):
+def construct_connect_inventory(hosts, credential, connection_port,
+                                exclude_hosts=None):
     """Create a dictionary inventory for Ansible to execute with.
 
     :param hosts: The collection of hosts to test connections
     :param credential: The credential used for connections
     :param connection_port: The connection port
-    :returns: A dictionary of the ansible invetory
+    :param exclude_hosts: Optional. Hosts to exclude test connections
+    :returns: A dictionary of the ansible inventory
     """
     inventory = None
     hosts_dict = {}
 
-    for host in hosts:
-        hosts_dict[host] = None
+    if exclude_hosts is None:
```

Comment:
I feel like we should be calculating what our finalized hosts are before we initializing them inside the `hosts_dict` so that we are not repeating code (`hosts_dict[host]=None`). 

1. Calculate hosts (do the set subtraction)
2. Loop through the finalized hosts/ add them to the `hosts_dict`

Reply:
1. Would be do the set subtraction if excludes is not None.